electrum/plugins/ledger/ledger.py

```diff
@@ -540,75 +764,548 @@ def sign_transaction(self, tx, password):
+    def get_master_fingerprint(self) -> bytes:
+        if self.master_fingerprint is None:
+            self.master_fingerprint = self.client.get_master_fingerprint()
+        return self.master_fingerprint
+
+    @runs_in_hwd_thread
+    @test_pin_unlocked
+    def get_singlesig_default_wallet_policy(self, addr_type: 'AddressType', account: int) -> 'WalletPolicy':
+        assert account >= HARDENED_FLAG
+
+        if addr_type == AddressType.LEGACY:
+            template = "pkh(@0/**)"
+        elif addr_type == AddressType.WIT:
+            template = "wpkh(@0/**)"
+        elif addr_type == AddressType.SH_WIT:
+            template = "sh(wpkh(@0/**))"
+        elif addr_type == AddressType.TAP:
+            template = "tr(@0/**)"
+        else:
+            raise ValueError("Unknown address type")
+
+        fpr = self.get_master_fingerprint()
+        key_origin_steps = f"{get_bip44_purpose(addr_type)}'/{get_bip44_chain(self.client.chain)}'/{account & ~HARDENED_FLAG}'"
+        xpub = self.get_xpub(f"m/{key_origin_steps}", 'standard')
+        key_str = f"[{fpr.hex()}/{key_origin_steps}]{xpub}"
+
+        # Make the Wallet object
+        return WalletPolicy(name="", descriptor_template=template, keys_info=[key_str])
+
+    @runs_in_hwd_thread
+    @test_pin_unlocked
+    def get_singlesig_policy_for_path(self, path: str, xtype: str, master_fp: bytes) -> Optional['WalletPolicy']:
+        path = path.replace("h", "'")
+        path_parts = path.split("/")
+
+        if not 5 <= len(path_parts) <= 6:
+            raise UserFacingException(f"Unsupported path: {path}")
+
+        path_root = "/".join(path_parts[:-2])
+
+        fpr = self.get_master_fingerprint()
+
+        # Ledger always uses standard xpubs in wallet policies
+        xpub = self.get_xpub(f"m/{path_root}", 'standard')
+
+        key_info = f"[{fpr.hex()}/{path_root}]{xpub}"
+
+        if xtype == 'p2pkh':
+            name = "Legacy P2PKH"
+            descriptor_template = "pkh(@0/**)"
+        elif xtype == 'p2wpkh-p2sh':
+            name = "Nested SegWit"
+            descriptor_template = "sh(wpkh(@0/**))"
+        elif xtype == 'p2wpkh':
+            name = "SegWit"
+            descriptor_template = "wpkh(@0/**)"
+        elif xtype == 'p2tr':
+            name = "Taproot"
+            descriptor_template = "tr(@0/**)"
+        else:
+            return None
+
+        policy = WalletPolicy("", descriptor_template, [key_info])
+        if is_policy_standard(policy, master_fp, constants.net.BIP44_COIN_TYPE):
+            return policy
+
+        # Non standard policy, so give it a name
+        return WalletPolicy(name, descriptor_template, [key_info])
+
+    def password_dialog(self, msg=None):
+        response = self.handler.get_word(msg)
+        if response is None:
+            return False, None, None
+        return True, response, response
+
+    def _register_policy_if_needed(self, wallet_policy: 'WalletPolicy') -> tuple[bytes, bytes]:
```

Comment:
`TypeError: 'type' object is not subscriptable`
[This syntax](https://peps.python.org/pep-0585/) was only added in python 3.9.
(Our minimum atm is 3.8)

Reply:
Indeed, thanks!
Fixed and rebased, crossing fingers for the CI run... :crossed_fingers: